sdk/typespec/typespec_client_core/src/async_runtime/mod.rs

```diff
@@ -189,12 +192,16 @@ pub fn set_async_runtime(runtime: Arc<dyn AsyncRuntime>) -> crate::Result<()> {
 }
 
 fn create_async_runtime() -> Arc<dyn AsyncRuntime> {
-    #[cfg(not(feature = "tokio"))]
+    #[cfg(all(target_arch = "wasm32", feature = "wasm-bindgen"))]
     {
-        Arc::new(standard_runtime::StdRuntime)
+        Arc::new(web_runtime::WasmBindgenRuntime) as Arc<dyn AsyncRuntime>
     }
     #[cfg(feature = "tokio")]
     {
         Arc::new(tokio_runtime::TokioRuntime) as Arc<dyn AsyncRuntime>
     }
+    #[cfg(not(any(feature = "tokio", feature = "wasm-bindgen")))]
+    {
+        Arc::new(standard_runtime::StdRuntime)
+    }
 }
```

Comment:
We should have a `panic!()` since there now exists a case we otherwise fall through: `target_arch == "wasm32"` and NOT `feature = "wasm-bindgen"`.

Reply:
Actually, I see that the `standard_runtime` module handles the `not(target_arch = "wasm32")` case itself, so ignore this comment.